cogs/handover.py

```diff
@@ -0,0 +1,154 @@
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def handover(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "handover" command.
+
+        The "handover" command runs the relavent handover methods
+        which will perform the following actions:
+        - Give @Committee role to anyone with @Committee-Elect
+        - Remove @Committee-Elect from anyone that has it
+        - Remove permissions for @Committee-Elect from all channels except #handover
+
+        In order to do this the bot will need to hold a role above that of the committee role.
+        """
+        committee_role: discord.Role = await self.bot.committee_role
+        committee_elect_role: discord.Role = await self.bot.committee_elect_role
+        main_guild: discord.Guild = self.bot.main_guild
+        handover_channel: discord.TextChannel = await self.bot.handover_channel
+
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Running the handover procedures... :hourglass:",
+        )
+        logger.debug("Running the handover command!")
+
+        for category in main_guild.categories:
+            logger.debug("Found category: %s", category.name.lower())
+            if "committee" in category.name.lower() and "archive" not in category.name.lower():
+                for channel in category.channels:
+                    logger.debug("Resetting channel permissions for channel: %s", channel)
+                    await channel.set_permissions(committee_elect_role, overwrite=None)
+
+        for member in committee_role.members:
+            logger.debug("Giving user: %s, access to #handover", member)
+            await handover_channel.set_permissions(
+                member,
+                read_messages=True,
+                send_messages=True,
+            )
+
+            logger.debug("Removing committee role from user: %s", member)
+            await member.remove_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+        for member in committee_elect_role.members:
+            logger.debug("Giving user: %s, the committee role.", member)
+            await member.add_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+            logger.debug("Removing committee-elect role from user: %s", member)
+            await member.remove_roles(
+                committee_elect_role,
+                reason=f"{ctx.user} used TeX Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(content=":white_check_mark: Handover procedure complete!")
+
+
+class ResetRolesCommandCog(TeXBotBaseCog):
```

Comment:
Part of this was I had originally intended this to include opt in roles but then remembered we don't reset opt in roles, are there any I've missed we usually reset?

Reply:
I don't think any have been missed. Perhaps `Postdoc`? Although that's just a guess because I don't really understand what student level that pertains to.